server/service/software.go

```diff
@@ -132,20 +133,20 @@ func (r getSoftwareResponse) error() error { return r.Err }
 func getSoftwareEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
 	req := request.(*getSoftwareRequest)
 
-	software, err := svc.SoftwareByID(ctx, req.ID, false)
+	software, err := svc.SoftwareByID(ctx, req.ID, req.TeamID, false)
 	if err != nil {
 		return getSoftwareResponse{Err: err}, nil
 	}
 
 	return getSoftwareResponse{Software: software}, nil
 }
 
-func (svc *Service) SoftwareByID(ctx context.Context, id uint, includeCVEScores bool) (*fleet.Software, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
+func (svc *Service) SoftwareByID(ctx context.Context, id uint, teamID *uint, includeCVEScores bool) (*fleet.Software, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.AuthzSoftwareInventory{TeamID: teamID}, fleet.ActionRead); err != nil {
```

Comment:
We may want to ask product if we need to keep this backward compatible by allowing global access to team-only users.  I'd prefer not to and chalk it up to a bug.